### PR TITLE
Add gateway & max_balance to wallets

### PR DESCRIPTION
**Pull request by ysv**

### CODE REVIEW

File: app/models/wallet.rb [hunk 1]
```diff
@@ -1,14 +1,17 @@
...
+  serialize :gateway, JSON
+
   include BelongsToCurrency
 
   validates :name, :address, presence: true
   validates :status, inclusion: { in: %w[active disabled] }
   validates :kind, inclusion: { in: %w[hot warm cold deposit] }
   validates :nsig, numericality: { greater_than_or_equal_to: 1, only_integer: true }
+  validates :max_balance, numericality: { greater_than_or_equal_to: 0, only_integer: true }
+  validates :gateway, length: { maximum: 1000 }
```
Comment (mod) [inline app/models/wallet.rb]: maybe valid json ?

File: app/models/wallet.rb [hunk 2]
```diff
@@ -30,8 +33,10 @@ def wallet_url
 #  address     :string(255)      not null
 #  kind        :string(32)       not null
 #  nsig        :integer
+#  max_balance :integer          default(0), not null
```
Comment (mod) [inline app/models/wallet.rb]: locate gateway before max_balance
change max_balance type: DECIMAL(32,16) (double check with amount in orders table)

File: config/templates/seed/wallets.yml.erb [hunk 3]
```diff
@@ -1,27 +1,81 @@
+- name:         Ethereum Deposit Wallet
+  currency_id:  eth
+  address:      '0x2b9fBC10EbAeEc28a8Fc10069C0BC29E45eBEB9C'
+  kind:         deposit
+  nsig:         1
+  max_balance:  0
+  status:       active
+  gateway:
+    client: ETH
```
Comment (mod) [inline config/templates/seed/wallets.yml.erb]: let's agree on values ETH in uppercase is confusing and look like a currency code
client: bitgo, geth-node, bitcoind-node, openware-gateway

File: config/templates/seed/wallets.yml.erb [hunk 4]
```diff
@@ -1,27 +1,81 @@
...
+  nsig:         1
+  max_balance:  0
+  status:       active
+  gateway:
+    client: BTC
+    options:
+      json_rpc_endpoint: http://user:password@127.0.0.1:18332
+
+- name:         Bitcoin Hot Wallet
   currency_id:  eth
```
Comment (dinesh-skyach) [inline config/templates/seed/wallets.yml.erb]: @ysv currency_id: btc

File: config/templates/seed/wallets.yml.erb (hunk 3 above)
Comment (mod) [reply config/templates/seed/wallets.yml.erb]: use DECIMAL for balance
max_balance:  0.00 if possible

File: config/templates/seed/wallets.yml.erb [hunk 5]
```diff
@@ -1,27 +1,81 @@
...
   address:      '0x2b9fBC10EbAeEc28a8Fc10069C0BC29E45eBEB9C'
   kind:         hot
   nsig:         1
+  max_balance:  100
   status:       active
+  gateway:
+    client: ETH
+    options:
+      json_rpc_endpoint: http://user:password@127.0.0.1:8545
```
Comment (mod) [inline config/templates/seed/wallets.yml.erb]: rename to endpoint it's not necessarly json-rpc

File: config/templates/seed/wallets.yml.erb (hunk 4 above)
Comment (ysv) [reply config/templates/seed/wallets.yml.erb]: @dinesh-skyach thanks

File: config/templates/seed/wallets.yml.erb [hunk 6]
```diff
@@ -1,27 +1,81 @@
...
 
 - name:         Ethereum Warm Wallet
   currency_id:  eth
   address:      '0x2b9fBC10EbAeEc28a8Fc10069C0BC29E45eBEB9C'
   kind:         warm
-  nsig:         2
-  status:       disabled
+  nsig:         1
+  max_balance:  100
```
Comment (mod) [inline config/templates/seed/wallets.yml.erb]: set the following default values
deposit eth: 0.00 max
hot eth: 20.00 max
warm : 200.00 max
cold : 10,000.00

**Review by mod — CHANGES_REQUESTED** [review]

**Pull request merged**